xarray/tests/test_dataarray.py

```diff
@@ -4282,6 +4282,10 @@ def test_polyfit(self, use_dask, use_datetime):
         ).T
         assert_allclose(out.polyfit_coefficients, expected, rtol=1e-3)
 
+        # Full output and deficient rank
+        out = da.polyfit("x", 12, full=True)
```

Comment:
Can you catch this warning as well - it ends up in the test output.

https://dev.azure.com/xarray/xarray/_build/results?buildId=3540&view=logs&j=acc1347f-8235-55b0-95c7-0e2189e61659&t=7284452c-b1fa-561e-b473-09e477cf0d79&l=442